### ISSUE TITLE
Useless chapters.

### ISSUE THREAD

**Issue by Gathros** [leios/algorithm-archive#33]
The AAA has a couple of useless chapters like Mathematical Background and Principle of Code. The Chapter A Personal Note is weird but I think it should be renamed preface and should be less personal.

**Comment by julianschacherpp** [comment]
To sum up, what I already said in Discord regarding that topic and the changes being in discussion (and to present my vision of implementing them):
I find the Mathematical Background chapter pretty useful and would combine it with another chapter, if needed. If we go full only-algorithm we can get rid of it for my part.
The Principle of Code chapter is something I would like to keep in a separate book or guide and link to it in an Introduction / HowToContribute page, to satisfy a more algorithm-focused book.
About the A Personal Note page: I really like it. It adds a personal touch to the project and I wouldn't  discard it.

**Comment by Gathros** [comment]
It's really long. BTW I haven't read it.

**Comment by julianschacherpp** [comment]
@Gathros The "A Personal Note"?

**Comment by Gathros** [comment]
Yes, it's better if it was shorter and renamed to prelude or something like that.

**Comment by julianschacherpp** [comment]
@Gathros The naming doesn't matter that much I think. Who wants to read it, does, who doesn't want to, doesn't. I read it and found it really interesting and lovely.

**Comment by Gathros** [comment]
If the goal of the AAA is to be a book then preludes would make sense since books have preludes.

**Comment by julianschacherpp** [comment]
@Gathros That's true. A personal note fits as well I think. It doesn't really matter what name you give to the prelude.

**Comment by leios** [comment]
@Gathros As it currently stands, I do not think there are many useless chapters, except for *A Personal Note*. The mathematical background and principles of code are obviously not the purpose of this book, but are kinda necessary for newcomers. I kept the material there super short and don't intend to make it any longer than necessary.

Some things, like Taylor Series and Convolutions belong in the mathematical background because they are used in a number of algorithms all over the place.

**Comment by julianschacherpp** [comment]
@leios I would keep _A Personal note_. It adds something to the AAA.

**Comment by leios** [comment]
As a note, the language discussion can probably be omitted; however, I really wanted to do language benchmarks later, which I feel would fit into AAA discussion pretty well. 

The AAA is not meant to be an introductory text, but we'll have a bunch of people who don't know how to code look into it because of how I advertise it on youtube. I think a ~10 page crash course to a bunch of different topics is fine and common in most books, it's just often skipped by the reader.

**Comment by Gathros** [comment]
I don't think a crash course is really need beyond the algorithms.

**Comment by leios** [comment]
Hmm. I see what you are saying. I will leave this issue open for a bit longer. We need to see what happens as new algorithms come in. The Mathematical Background chapter was basically meant as a place to store concepts needed to understand future algorithms. The Principles of Code section was just there to make sure everyone knew the concepts we were going to be using and have an introduction to compiling the languages used in the archive.

That said, the Principles of Code could easily be split into a chapter on contributing to the AAA (with just the version control chapter), and another one on benchmarking (when we get there).

**Comment by leios** [comment]
My goal is to tackle this this weekend. I will re-write the *Principles of Code* chapter as a general Data structures chapter with information on the data structures already used in the archive. I'll keep updating this  with time.

The language selection will be removed, but I still want to eventually add it back.

I will also move the version control section into a general *How to Contribute* section (which I realize will also be updated soon)
